### PR TITLE
Redirect

### PR DESCRIPTION
**Pull request by amandate**
# Issue/Task:
This should redirect user input to call the correct method in `commands.py`

I also added some additional methods, since we decided to generalize some commands:
* delete food storage
* delete freezer
* delete fridge
* open food storage
* open freezer
* open fridge

# Checklist
- [x] Fixed Issue/Accomplished Task
- [x] Wrote tests
- [x] All test cases passing

### CODE REVIEW

File: src/Session/session.py [hunk 1]
```diff
@@ -1,36 +1,90 @@
...
+            user_input == OPEN_FREEZER or \
+            user_input == OPEN_FRIDGE:
+            self._commands.open(SPACE.join(user_input.split()[1:]))
+        elif user_input == REMOVE_FOOD:
+            self._commands.remove_food()
+        elif user_input == SAVE:
+            self._commands.save()
+        else:
+            print(INVALID_RESPONSE_MESSAGE)
+        return 1
```
Comment (leezahuor) [inline src/Session/session.py:78]: This is so cool!!

File: src/Session/session.py [hunk 2]
```diff
@@ -1,36 +1,90 @@
...
-                break
-
-    def end(self):
-        pass 
-
-
+            user_input = input(ENTER_COMMAND_MESSAGE).strip()
+            success = self._redirect(user_input)
+            if not success:
+                break
```
Comment (leezahuor) [inline src/Session/session.py:90]: This too!

File: tests/Session/test_session.py [hunk 3]
```diff
@@ -1,14 +1,92 @@
...
-        # TODO: This can't be tested right now until we do regression tests
-        pass
\ No newline at end of file
+        # This skips checking the welcome message, since twoPrintOutcomes
+        # has been tested in utils and it depends on someone's personal profiles.
+
+        # Testing that the 'exit' command properly exits the program
+        mocked_input.side_effect = [EXIT]
+        self.session.start()
+        self.assertRaises(SystemExit)
```
Comment (leezahuor) [inline tests/Session/test_session.py:55]: Do you think you can explain how `assertRaises()` works? I tried looking it up in the documentation and it said it's to "Test that an exception is raised when callable is called with any positional or keyword arguments that are also passed to [assertRaises()] (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)." which I don't really understand. Is exiting an 'exception' in this case?

File: tests/Session/test_session.py [hunk 4]
```diff
@@ -1,14 +1,92 @@
...
+        self.checkCommandMethodCalled(mocked_input, [HELP, EXIT], command_class_path + ".help")
+        self.checkCommandMethodCalled(mocked_input, [LIST_FOOD_STORAGES, EXIT], command_class_path + ".list_food_storages", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [LIST_FREEZERS, EXIT], command_class_path + ".list_food_storages", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [LIST_FRIDGES, EXIT], command_class_path + ".list_food_storages", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [LOAD, EXIT], command_class_path + ".load")
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FOOD_STORAGE, EXIT], command_class_path + ".open", FOOD_STORAGE)
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FREEZER, EXIT], command_class_path + ".open", FREEZER)
+        self.checkCommandMethodCalled(mocked_input, [OPEN_FRIDGE, EXIT], command_class_path + ".open", FRIDGE)
+        self.checkCommandMethodCalled(mocked_input, [REMOVE_FOOD, EXIT], command_class_path + ".remove_food")
+        self.checkCommandMethodCalled(mocked_input, [SAVE, EXIT], command_class_path + ".save")
```
Comment (leezahuor) [inline tests/Session/test_session.py:77]: I like how you did this, very cool.

**Review by leezahuor — COMMENTED** [review]
I just need some clarification with how the testing part works, but other than that everything looks really good!

File: tests/Session/test_session.py (hunk 3 above)
Comment (amandate) [reply tests/Session/test_session.py:55]: Yeah! I haven't read the documentation, but essentially it's checking that a specific exception is raised. So here, I'm checking that the SystemExit Exception is raised. 

In our case, this happens when the program ends from:
1. the user presses keys: `ctrl` + `c`
2. they type 'exit', which makes us break out of the while loop in `start()`, stopping naturally

File: tests/Session/test_session.py (hunk 4 above)
Comment (amandate) [reply tests/Session/test_session.py:77]: thank you! it actually took me a long time to figure out how to get it to work lol i was having a hard time figuring out how to check that a certain command was called. Let me know if you need help understanding this, I know it's a lot!

File: tests/Session/test_session.py (hunk 3 above)
Comment (leezahuor) [reply tests/Session/test_session.py:55]: I see, thank you!!

**Review by leezahuor — APPROVED** [review]
Thanks for the explanations! Everything looks great!

**Pull request merged**